include/field.hxx

```diff
@@ -136,6 +136,14 @@ class Field {
     return directions.z;
   }
 
+  /// Setters for *DirectionType
+  void setDirectionY(YDirectionType y_type) {
+    directions.y = y_type;
+  }
+  void setDirectionZ(ZDirectionType z_type) {
+    directions.z = z_type;
```

Comment:
Can we just make `directions` public? Currently the getters and setters are trivial, so there's no invariant we need to preserve. Are we likely to add one in the future such that these might be less trivial?

Reply:
This does seem weird. Originally I think I'd hoped not to need the setters at all, but now that seems inconvenient at least. I think the setters in `Field` should go: there are setters in `Field3D`/`Field2D`/`FieldPerp` that are used for method-chaining like `auto result{emptyFrom(f).setDirectionY(YDirectionType::Aligned)}`. At the moment they call `Field::setDirectionY`, etc., but there's no need for them to. I think at one point we had a general `Direction` type, and the setters were checking that it was a valid y-direction, but that's gone now. I just noticed that `Field2D` is missing the setters, which it should probably have in case we want them in a template function at some point. I suggest I make a separate PR to tidy up `setDirectionY` and `setDirectionZ`.